code/game/gamemodes/cult/runes.dm

```diff
@@ -14,1131 +14,1115 @@ var/list/sacrificed = list()
 
 
 /////////////////////////////////////////FIRST RUNE
-	proc
-		teleport(var/key)
-			var/mob/living/user = usr
-			var/allrunesloc[]
-			allrunesloc = new/list()
-			var/index = 0
-		//	var/tempnum = 0
-			for(var/obj/effect/rune/R in world)
-				if(R == src)
-					continue
-				if(R.word1 == cultwords["travel"] && R.word2 == cultwords["self"] && R.word3 == key && isPlayerLevel(R.z))
-					index++
-					allrunesloc.len = index
-					allrunesloc[index] = R.loc
-			if(index >= 5)
-				user << "<span class='danger'>You feel pain, as rune disappears in reality shift caused by too much wear of space-time fabric.</span>"
-				if (istype(user, /mob/living))
-					user.take_overall_damage(5, 0)
-				qdel(src)
-			if(allrunesloc && index != 0)
-				if(istype(src,/obj/effect/rune))
-					user.say("Sas[pick("'","`")]so c'arta forbici!")//Only you can stop auto-muting
-				else
-					user.whisper("Sas[pick("'","`")]so c'arta forbici!")
-				user.visible_message("<span class='danger'>[user] disappears in a flash of red light!</span>", \
-				"<span class='danger'>You feel as your body gets dragged through the dimension of Nar-Sie!</span>", \
-				"<span class='danger'>You hear a sickening crunch and sloshing of viscera.</span>")
-				user.loc = allrunesloc[rand(1,index)]
-				return
-			if(istype(src,/obj/effect/rune))
-				return	fizzle() //Use friggin manuals, Dorf, your list was of zero length.
-			else
-				call(/obj/effect/rune/proc/fizzle)()
-				return
-
-
-		itemport(var/key)
-//			var/allrunesloc[]
-//			allrunesloc = new/list()
-//			var/index = 0
-		//	var/tempnum = 0
-			var/culcount = 0
-			var/runecount = 0
-			var/obj/effect/rune/IP = null
-			var/mob/living/user = usr
-			for(var/obj/effect/rune/R in world)
-				if(R == src)
-					continue
-				if(R.word1 == cultwords["travel"] && R.word2 == cultwords["other"] && R.word3 == key)
-					IP = R
-					runecount++
-			if(runecount >= 2)
-				user << "<span class='danger'>You feel pain, as rune disappears in reality shift caused by too much wear of space-time fabric.</span>"
-				if (istype(user, /mob/living))
-					user.take_overall_damage(5, 0)
-				qdel(src)
-			for(var/mob/living/carbon/C in orange(1,src))
-				if(iscultist(C) && !C.stat)
-					culcount++
-			if(culcount>=3)
-				user.say("Sas[pick("'","`")]so c'arta forbici tarem!")
-				user.visible_message("<span class='warning'>You feel air moving from the rune - like as it was swapped with somewhere else.</span>", \
-				"<span class='warning'>You feel air moving from the rune - like as it was swapped with somewhere else.</span>", \
-				"<span class='warning'>You smell ozone.</span>")
-				for(var/obj/O in src.loc)
-					if(!O.anchored)
-						O.loc = IP.loc
-				for(var/mob/M in src.loc)
-					M.loc = IP.loc
-				return
-
-			return fizzle()
+/obj/effect/rune/proc/teleport(var/key)
```

Comment:
Can you refactor all of these procs to perhaps accept a `/mob/living/user` argument as the first argument, and then pass it in there? `usr` should not be used at this level ever.